src/clj/fluree/db/json_ld/transact.cljc

```diff
@@ -16,7 +17,12 @@
        (into []
              (comp (remove (fn [[k _]] (keyword? k)))  ; remove :id :idx :type
                    (mapcat rest)                      ; discard keys
-                   (mapcat (partial remove :value)))) ; remove value objects
+                   (mapcat (partial remove
+                                    (fn [v]
+                                      ;; remove value objects unless they have type xsd:anyURI
+                                      (and
+                                        (some? (:value v))
+                                        (not= (:type v) const/iri-anyURI)))))))
```

Comment:
Do we want to support both `@id` and `xsd:anyURI`? Or just `@id`? And if we're using `@id`, do we want to change the values clause syntax to allow the same method in addition to `xsd:anyURI` (or in place of it, though that breaks backwards compat).

Reply:
Just a personal anecdote for consideration:
When I was working with Paula on building out a small ontology for the Jarvis demo she told me the use of `xsd:anyURI` is a bit of a code smell as it should really only be used in the case of linking to videos, images, and other online resources. The places where I was using them, specifically "I want this property to be a reference to another node", she advised me to use [`rdfs:Resource`](https://www.w3.org/TR/rdf-schema/#ch_resource) instead.
I may be mixing concepts at this point, but this makes me think we shouldn't lean on `xsd:anyURI`.

Reply:
@dpetran I don't think we should use `xsd:anyURI` to describe the data type of an iri literal at all in user facing apis (including values clauses) because it's not techically correct. Neither is `@id`, but that's more correct than `xsd:anyURI` and we've got to do something

@Jackamus29 yep, as usual, Paula is right. That's the impetus for making this change. I don't like using rdfs:Resource either at first blush because we don't want to reference another node here; we'd like to indicate that a bit of data is actually an iri and not a string in the same way use use `xsd:int` to indicate that the literal `"3"` is an integer and not a string or long, for example.